### ISSUE TITLE
RFE: please implement DefautInstance mechanism for systemctl start as well

### ISSUE THREAD

**Issue by Zugschlus** [systemd/systemd#4405]
In a template unit, I can use DefaultInstance to control which instance is actually enabled when I just systemctl enable unitname instead of systemctl enable unitname@instancename.

I would like to have a similiar mechanism for systemctl start.

If I type systemctl start unitname, and there is a unitname@.service but no unitname.service, I would like to be able to specify inside of unitname@.service which instance is started instead of complaining about the non-existing unitname.service.

This would be incredibly helpful for distributions when a service is used in just one instance in the vast majority of cases, and thus users expect to be able to use systemctl start foo instead of systemctl start foo@default to interact with the single instance of the service, while simultaneously allowing instantiated usage of the service without having to modify the unit from the package for the exotic, instantiated case.

I have tried writing a sidekick unit with ExecStart=/bin/true with dependencies to bring up foo@default with it and take it down, but was not successful in doing so. Even the mailing lists, Usenet groups and systemd-knowing acquaintainces failed in writing an appropriate sidekick unit.

If the task can be accomplished by writing an appropriate sidekick unit, then the docs are inadequate since noone in my filter bubble grokked the docs successfully to do this. In this case, this issue should be taken as a bug report regarding the docs. Even then, the sidekick unit is just a workaround for a missing StartDefaultUnit setting.

**Comment by mbiebl** [comment]
Can you define what exact behaviour you expect from the helper unit foo.service regarding its interaction with foo@bar.service and vice versa.

**Comment by Zugschlus** [comment]
systemctl start foo => foo@bar.service gets started
systemctl stop foo => foo@bar.service get stopped
If this RFE gets implemented, setting StartDefaultInstnace=bar in foo@.service would cause the same behavior without having an explicit foo.service Unit.

**Comment by thx1111** [comment]
I also consider this a good idea.  It is a little unexpected, discovering that DefaultInstance= does not currently apply to the start, stop, and restart commands.

I don't know that an additional StartDefaultUnit dependency is needed, where the existing DefaultInstance dependency would otherwise then be expected to apply to the enable and disable commands, and then also to the start, stop, and restart commands.

**Comment by Zugschlus** [comment]
DefaultInstance is in the [Install] section of a unit file, and the [Install] section is documented as "used by the enable and disable commands", so its behavior is consistent. My gut feeling says that a directive that controls start and stop belongs in the [Service] section.

**Comment by keszybz** [comment]
Yeah, there's an inconsistency here. In particular, we now have `systemctl enable --now …`, which (I presume) starts the default instance when used with a template name.

If this is implemented, I'd rather not have a separate StartDefaultUnit setting, in particular because `systemctl enable --now` would then start a different unit than `systemctl start` which I think would be too confusing. I also don't see much use in having the "install" and "start" DefaultInstances differ.

So yeah, I think 'systemctl start' should just fetch the DefaultInstance from [Install] section.

**Comment by rhysperry111** [comment]
Been running into this today with `enable --now`... is there likely going to be any change?
